articles/frontend-is-not-your-enemy.md

```diff
@@ -0,0 +1,157 @@
+
+## Compile Time Checking
+
+Ok, I know who you are... You are one of those people who call themselves *back-end* developer. And you probably worked in strongly typed languages, and you are used to having your IDE insult you with ***Cannot assign boolean to String!***  So you find frontend rather disgusting.
+
+Well guess what, you have it all now! You have it with TypeScript (or Flow, KotlinJS, Elm...). You can even use tslint.json to define the guidelines which the team will agree upon, and engineers will be notified if they for example define a method without a return type.
+
+```json
+// tslint.json
+{
+    "rules": {
+        "no-any": true,
+        "triple-equals": true,
+        "prefer-const": true
+    }
+}
+```
+
+You can check all available tslint rules here - <https://palantir.github.io/tslint/rules/>
+
+Now that we have our simple tslint, let's see how it works:
+
+```ts
+// some-file.ts
+
+let me: any = "John Doe"; // Error - cannot use 'any', use 'string'
+
+me = false; // Error - cannot assign boolean to string variable.
+
+const age = 17;
+
+age = 18; // Error - cannot reassign const value
+
+let isEighteen = age == 18; // Warning - Use === instead of ==
+
+// Warning - `isEighteen` should be a const
+
+```
+
+You are probably thinking: "Ok well const is ES6, my browser will tell me i cannot reassign it". That is correct, but now, you will be notified at compile-time, meaning your app won't build until you fix the issue. Thus, we have strictly typed JavaScript app.
+
+It has all the OO principles that Java has (and many much more sophisticated), such as interfaces, abstract classes, enums, and many many more.
+
+If this catches your attention, I advise to visit https://www.typescriptlang.org/docs/handbook/basic-types.html for more detailed information.
+
+## Build Tools and Node
+
+Yes, you are on the same article. There's proper build in frontend. For a while now... And hear this, one can even import .css in .js files now! I know, what kind of monster would do such a thing. Well...
+
+**Node** basically opened the door to a whole new universe of magic we can do in front-end development. And do so easily.
+
+It started with simple task-runners that can help you organize your code, clean-up, uglify/minify it, and even transpile it to something different.
+
+First tools worth mentioning was **Grunt** and **Gulp**, with much nicer syntax. Now we have **Webpack**, and just recently, version 4 was released. By the time you are reading this, it is probable that v5.0 is out. Also, there's a new Webpack nemesis called Parcel. It's in a hype phase now and many think it will overtake Webpack in upcoming years...
+
+Regardless of the tools you prefer, it is suggested to, when deciding on the tool, do thorough analysis, and choose the one which is battle tested, has a great community support, is feature-rich, and for complex project - is it performant.
+
+## Frameworks and Libraries
```

Comment:
Maybe to say here about difficulty to choose a framework in a large ocean of those.